### ISSUE TITLE
add support for PyTorch's functional modules

### ISSUE THREAD

**Issue by shaymeister** [shaymeister/parcae#3]
# Convolution Functions
- [ ] torch.nn.functional.conv1d
- [ ] torch.nn.functional.conv2d
- [ ] torch.nn.functional.conv3d
- [ ] torch.nn.functional.conv_transpose1d
- [ ] torch.nn.functional.conv_transpose2d
- [ ] torch.nn.functional.conv_transpose3d
- [ ] torch.nn.functional.unfold
- [ ] torch.nn.functional.fold

# Pooling functions
- [ ] torch.nn.functional.avg_pool1d
- [ ] torch.nn.functional.avg_pool2d
- [ ] torch.nn.functional.avg_pool3d
- [ ] torch.nn.functional.max_pool1d
- [ ] torch.nn.functional.max_pool2d
- [ ] torch.nn.functional.max_pool3d
- [ ] torch.nn.functional.max_unpool1d
- [ ] torch.nn.functional.max_unpool2d
- [ ] torch.nn.functional.max_unpool3d
- [ ] torch.nn.functional.lp_pool1d
- [ ] torch.nn.functional.lp_pool2d
- [ ] torch.nn.functional.adaptive_max_pool1d
- [ ] torch.nn.functional.adaptive_max_pool2d
- [ ] torch.nn.functional.adaptive_max_pool3d
- [ ] torch.nn.functional.adaptive_avg_pool1d
- [ ] torch.nn.functional.adaptive_avg_pool2d
- [ ] torch.nn.functional.adaptive_avg_pool3d
- [ ] torch.nn.functional.fractional_max_pool2d
- [ ] torch.nn.functional.fractional_max_pool3d

# Non-linear activation functions
- [ ] torch.nn.functional.threshold
- [ ] torch.nn.functional.threshold_
- [ ] torch.nn.functional.relu
- [ ] torch.nn.functional.relu_
- [ ] torch.nn.functional.hardtanh
- [ ] torch.nn.functional.hardtanh_
- [ ] torch.nn.functional.hardswish
- [ ] torch.nn.functional.relu6
- [ ] torch.nn.functional.elu
- [ ] torch.nn.functional.elu_
- [ ] torch.nn.functional.selu
- [ ] torch.nn.functional.celu
- [ ] torch.nn.functional.leaky_relu
- [ ] torch.nn.functional.leaky_relu_
- [ ] torch.nn.functional.prelu
- [ ] torch.nn.functional.rrelu
- [ ] torch.nn.functional.rrelu_
- [ ] torch.nn.functional.glu
- [ ] torch.nn.functional.gelu
- [ ] torch.nn.functional.logsigmoid
- [ ] torch.nn.functional.hardshrink
- [ ] torch.nn.functional.tanhshrink
- [ ] torch.nn.functional.softsign
- [ ] torch.nn.functional.softplus
- [ ] torch.nn.functional.softmin
- [ ] torch.nn.functional.softmax
- [ ] torch.nn.functional.softshrink
- [ ] torch.nn.functional.gumbel_softmax
- [ ] torch.nn.functional.log_softmax
- [ ] torch.nn.functional.tanh
- [ ] torch.nn.functional.sigmoid
- [ ] torch.nn.functional.hardsigmoid
- [ ] torch.nn.functional.silu
- [ ] torch.nn.functional.mish
- [ ] torch.nn.functional.batch_norm
- [ ] torch.nn.functional.group_norm
- [ ] torch.nn.functional.instance_norm
- [ ] torch.nn.functional.layer_norm
- [ ] torch.nn.functional.local_response_norm
- [ ] torch.nn.functional.normalize

# Linear functions
- [ ] torch.nn.functional.linear
- [ ] torch.nn.functional.bilinear
- [ ] torch.nn.functional.dropout
- [ ] torch.nn.functional.alpha_dropout
- [ ] torch.nn.functional.feature_alpha_dropout
- [ ] torch.nn.functional.dropout2d
- [ ] torch.nn.functional.dropout3d
- [ ] torch.nn.functional.embedding
- [ ] torch.nn.functional.embedding_bag
- [ ] torch.nn.functional.one_hot

# Distance functions
- [ ] torch.nn.functional.pairwise_distance
- [ ] torch.nn.functional.cosine_similarity
- [ ] torch.nn.functional.pdist
	
# Loss functions
- [ ] torch.nn.functional.binary_cross_entropy
- [ ] torch.nn.functional.binary_cross_entropy_with_logits
- [ ] torch.nn.functional.poisson_nll_loss
- [ ] torch.nn.functional.cosine_embedding_loss
- [ ] torch.nn.functional.cross_entropy
- [ ] torch.nn.functional.ctc_loss
- [ ] torch.nn.functional.gaussian_nll_loss
- [ ] torch.nn.functional.hinge_embedding_loss
- [ ] torch.nn.functional.kl_div
- [ ] torch.nn.functional.l1_loss
- [ ] torch.nn.functional.mse_loss
- [ ] torch.nn.functional.margin_ranking_loss
- [ ] torch.nn.functional.multilabel_margin_loss
- [ ] torch.nn.functional.multilabel_soft_margin_loss
- [ ] torch.nn.functional.multi_margin_loss
- [ ] torch.nn.functional.nll_loss
- [ ] torch.nn.functional.huber_loss
- [ ] torch.nn.functional.smooth_l1_loss
- [ ] torch.nn.functional.soft_margin_loss
- [ ] torch.nn.functional.triplet_margin_loss
- [ ] torch.nn.functional.triplet_margin_with_distance_loss

# Vision functions
- [ ] torch.nn.functional.pixel_shuffle
- [ ] torch.nn.functional.pixel_unshuffle
- [ ] torch.nn.functional.pad
- [ ] torch.nn.functional.interpolate
- [ ] torch.nn.functional.upsample
- [ ] torch.nn.functional.upsample_nearest
- [ ] torch.nn.functional.upsample_bilinear
- [ ] torch.nn.functional.grid_sample
- [ ] torch.nn.functional.affine_grid
